library/ssl_client.c

```diff
@@ -714,6 +714,50 @@ static int ssl_generate_random( mbedtls_ssl_context *ssl )
     return( ret );
 }
 
+static int ssl_session_set_hostname( mbedtls_ssl_session *session,
```

Comment:
An "#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION)" guard is probably missing and we should probably factorize the code of this function and that of `mbedtls_ssl_set_hostname()` in ssl_tls.c.